modules/container/src/main/java/org/mule/runtime/container/api/TestContainerClassLoaderAssembler.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.container.api;
+
+import org.mule.runtime.container.internal.DefaultTestContainerClassLoaderAssembler;
+import org.mule.runtime.module.artifact.api.classloader.ArtifactClassLoader;
+import org.mule.runtime.module.artifact.api.classloader.ClassLoaderLookupPolicy;
+
+import java.net.URL;
+import java.util.List;
+import java.util.Set;
+
+public interface TestContainerClassLoaderAssembler extends AutoCloseable {
+
+  static TestContainerClassLoaderAssembler create(List<String> extraBootPackages, Set<String> extraPrivilegedArtifacts,
+                                                  List<URL> urls) {
+    return new DefaultTestContainerClassLoaderAssembler(extraBootPackages, extraPrivilegedArtifacts, urls);
+  }
+
+  ArtifactClassLoader createContainerClassLoader();
+
+  ArtifactClassLoader getContainerClassLoader();
```

Comment:
what would be the difference between these 2?

Reply:
I ended up removing this in favour of encapsulating the usage of the `ContainerOnlyLookupStrategy`.